### ISSUE TITLE
Chapitre 4 : SGDRegressor - DG Ordinaire et Stochastique 

### ISSUE THREAD

**Issue by mlm-root** [ageron/handson-ml2#268]
Bonjour

D'après le tableau "4.1 Comparaison d'algorithmes de regression linéaire", il est possible d'entrainer un modèle à l'aide d'une descente de gradient ordinaire via le SGDRegressor. J'avoue ne pas comprendre puisque, par définition, le SGDRegressor s'appuie sur la descente de gradient stochastique. 

Selon moi, le SGDRegressor, à l'appel de la fonction fit, détermine de nouveaux paramètres via le calcul des dérivées partielles pour une observation, choisie au hasard, et issue du jeu d'entrainement fourni à la fonction fit. L'opération est réitérée max_iter fois.

Ma question, comment peut-on "dire" au SGDRegressor d'utiliser TOUTES les observations du jeu d'entrainement lors du calcul des dérivées partielles ? (même question pour la DG mini-batch finalement).  

Peut-être que la réponse se trouve dans les hyperparamètres passés au SGDRegressor mais malgré mes recherches je ne trouve pas... :(

Merci pour votre aide 
Melvin

**Comment by ibeauregard** [comment]
The following answer appears to confirm your concerns : https://stackoverflow.com/questions/55552139/sklearn-implementation-for-batch-gradient-descend

**Comment by mlm-root** [comment]
Thank you @ibeauregard
